keps/sig-network/4444-service-routing-preference/README.md

```diff
@@ -0,0 +1,998 @@
+  * **Description:** Confirms that the dataplane has received the hints,
+    acknowledged them, and configured itself accordingly.
+
+#### Condition usage by other implementations
+
+Other implementations supporting `routingPreference` **should** adopt a domain
+prefixing strategy for their condition types. This means prefixing condition
+types with a domain string (e.g., `my.domain.io/RoutingPreferenceAccepted`) to
+prevent conflicts when multiple control planes (like the default EndpointSlice
+controller) are present.
+
+<<[UNRESOLVED Name for the field is being discussed]>>
+
+### Choice of field name
+The name `routingPreference` is meant to capture the highly
+implementation-specific nature of this field and how it affects the routing of
+traffic
+* Field names that include the word "topology", like `topologyPreference` and
+  `topologyRoutingPreference`, were avoided because:
+  * Topology word might be a bit vague
+  * The actual heuristic might not specifically be tied to the geographical or
+    network topology, but might also be considering factors like latency, active
+    connections, cpu usage of the backends, etc.
+* Use of the word “hint” for names like `trafficRoutingHint` and
+  `customRoutingHint` was avoided so as not to confuse with the hint field
+  within EndpointSlices which has previously been discussed in relation to such
+  routing heuristics. In a sense, the hint field within EndpointSlices is just
+  an implementation detail.
+* Use of words like “traffic” and “policy” for names like
+  trafficRoutingPreference was not favored so that it’s clearly different from
+  the `internalTrafficPolicy` and `externalTrafficPolicy` fields which have a
+  more fixed behavior.
+* Use of the word "selection" for a name like `endpointSelection` were avoided
+  so as not to confuse with the actual process of selecting the complete set of
+  pods backing a service.
+
+<<[/UNRESOLVED]>>
+
+### Intersection with internal/externalTrafficPolicy
+
+The intersection of the field with `internalTrafficPolicy` and
+`externalTrafficPolicy` fields remains the same as the annotation. The following
+table borrowed from [KEP-2086: Service Internal Traffic
+Policy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2086-service-internal-traffic-policy)
+captures the precedence
+
+<table>
+<thead>
+  <tr>
+    <th>ExternalTrafficPolicy</th>
+    <th>InternalTrafficPolicy</th>
+    <th>routingPreference</th>
+    <th>External Result</th>
+    <th>Internal Result</th>
+  </tr>
+</thead>
+<tbody>
+  <tr>
+    <td>-</td>
+    <td>-</td>
+    <td>Auto</td>
+    <td>routingPreference=Auto</td>
+    <td>routingPreference=Auto</td>
+  </tr>
+  <tr>
+    <td>Local</td>
+    <td>-</td>
+    <td>Auto</td>
+    <td>ExternalTrafficPolicy=Local</td>
+    <td>routingPreference=Auto</td>
+  </tr>
+  <tr>
+    <td>Local</td>
+    <td>Local</td>
+    <td>Auto</td>
+    <td>ExternalTrafficPolicy=Local</td>
+    <td>InternalTrafficPolicy=Local</td>
```

Comment:
```suggestion
    <td>routingPreference: Auto</td>
    <td>routingPreference: Auto</td>
  </tr>
  <tr>
    <td>Local</td>
    <td>-</td>
    <td>Auto</td>
    <td>ExternalTrafficPolicy: Local</td>
    <td>routingPreference: Auto</td>
  </tr>
  <tr>
    <td>Local</td>
    <td>Local</td>
    <td>Auto</td>
    <td>ExternalTrafficPolicy: Local</td>
    <td>InternalTrafficPolicy: Local</td>
```

This is valid YAML (snippets) and is closer to valid JSON than the version using `=`.